### ISSUE TITLE
False positive for `no-invalid-position-declaration` inside a mixin

### ISSUE THREAD

**Issue by Vinnl** [stylelint-scss/stylelint-config-recommended-scss#369]
We have the following in our code, which used to be fine, but now gets flagged for `no-invalid-position-declaration` after updating to the latest versions:

```scss
@mixin company-name-area {
  @media screen and (min-width: $screen-lg) {
    min-width: 200px;
  }
}
```

As far as I can see this should be valid (though admittedly, I haven't really used SCSS mixins myself before). It looks similar to https://github.com/stylelint-scss/stylelint-config-recommended-scss/issues/364.

**Comment by jrmlt** [comment]
I'll propose a PR after https://github.com/stylelint/stylelint/pull/8781 is merged.
